region/norfair/crocomire.json

```diff
@@ -580,6 +564,69 @@
         {
           "from": 1,
           "to": [
+            {
+              "id": 1,
+              "strats": [
+                {
+                  "name": "Leave with Runway",
+                  "notable": false,
+                  "requires": [],
+                  "exitCondition": {
+                    "leaveWithRunway": {
+                      "length": 3,
+                      "openEnd": 1
+                    }
+                  }
+                }
+              ]
+            },
+            {
+              "id": 4,
+              "strats": [
+                {
+                  "name": "Enter Shinesparking",
+                  "notable": false,
+                  "entranceCondition": {
+                    "comeInWithSpark": {}
+                  },
+                  "requires": [
+                    {"shinespark": {
+                      "frames": 43,
+                      "excessFrames": 8
+                    }}
+                  ]
+                },
+                {
+                  "name": "Short Shinespark",
+                  "notable": false,
+                  "entranceCondition": {
+                    "comeInShinecharged": {"framesRequired": 45}
+                  },
+                  "requires": [
+                    {"shinespark": {
+                      "frames": 29,
+                      "excessFrames": 5
+                    }}
+                  ]
+                },
+                {
+                  "name": "SpaceJump across and Leave with Spark",
```

Comment:
This strat has made me think about how the door needs to be blue. I wonder how we should represent that, and/or if `door unlocked at node` is supposed to imply that. (This is likely a problem all over the place)

Reply:
Using a super on a green or pink door would work.  But broadly, yeah, colored doors are going to cost extra heat frames and shinespark timer frames that aren't being accounted for.  
There could be a time cost on ammo doors.  Represented either by heatFrame count + making you do the exitCondition strat again.  Or directly taking/adding to the heatFrame/ShineFrame counts of strats.  A lot of the time logic would except unlock door then refill and come back.

Reply:
The randomizer does include heat frame costs for opening locked doors. It's using 50 frames to open a pink door with Missiles and 110 frames for a yellow door (and nothing for using a Super). The code for that is here: https://github.com/blkerby/MapRandomizer/blob/cbf93f8949c93d54efae0d6e1410addcc5f278b1/rust/src/randomize.rs#L256

It's a little primitive at the moment, in that it doesn't yet treat ammo door locks as flags, so the cost has to be paid every time when using the door. Also these are essentially worst-case costs; in many cases the heat frames wouldn't need to be this large. So we might want to have some way to model these explicitly in the data, and/or document how the implicit behavior should work.